scripts/usb_reset.py

```diff
@@ -14,46 +14,266 @@
+        dev = read_int("/sys/bus/usb/devices/{}/devnum".format(device))
+        return "/dev/bus/usb/{0:03d}/{1:03d}".format(bus, dev)
+    except (IOError, ValueError) as e:
+        log.debug("Failed to get device path {}: {}".format(device, str(e)))
+        exit(1)
 
 
-if __name__ == "__main__":
-    log.logToSyslog(level=logging.DEBUG)
+def get_ctl(path, mode):
+    try:
+        st = os.stat(path)
+    except OSError:
+        return None
+    t = ""
+    if S_ISBLK(st.st_mode):
+        t = "b "
+    elif S_ISCHR(st.st_mode):
+        t = "c "
+    if t and (mode == "r" or mode == "rw"):
+        return t + str(os.major(st.st_rdev)) + ":" + str(os.minor(
+            st.st_rdev)) + " " + mode
+    return None
+
+
+def device_ctl(path, dom_id, allow):
+    cg_dir = get_cg_dir(dom_id)
+    file_name = "devices.allow" if allow else "devices.deny"
+    try:
+        with open(cg_dir + file_name, "w", 0) as f:
+            ctl = get_ctl(path, "rw")
+            if ctl:
+                f.write(ctl)
+    except IOError as e:
+        # log and continue
+        log.debug("Failed to {} {}: {}".format(
+            "allow" if allow else "deny", path, str(e)))
+
+
+def setup_cgroup(dom_id, pid):
+    cg_dir = get_cg_dir(dom_id)
+
+    try:
+        os.mkdir(cg_dir, 0755)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            log.debug("Failed to create cgroup: {}".format(cg_dir))
+            exit(1)
+
+    try:
+        with open(cg_dir + "tasks", "w", 0) as tasks, \
+                open(cg_dir + "devices.deny", "w", 0) as deny, \
+                open(cg_dir + "devices.allow", "w", 0) as allow:
+
+            # deny all by default
+            deny.write("a")
 
-    if len(sys.argv) != 2:
-        sys.exit("usage: {} device_node".format(sys.argv[0]))
+            # Only the "/dev/null" is required after CP-25630
+            whitelist = [
+                ("/dev/null", "rw")
+            ]
+            for dev in whitelist:
+                ctl = get_ctl(*dev)
+                if ctl:
+                    allow.write(ctl)
 
-    device = sys.argv[1]
-    pattern = re.compile(r"^\d+-\d+(\.\d+)*$")
-    if pattern.match(device) is None:
-        log.debug("unexpected device node: {}".format(device))
+            tasks.write(str(pid))
+
+    except IOError as e:
+        log.debug("Failed to setup cgroup: {}".format(str(e)))
         exit(1)
 
+
+def mount(source, target, fs, flags=0):
+    if ctypes.CDLL(ctypes.util.find_library("c"), use_errno=True
+                   ).mount(source, target, fs, flags) < 0:
+        # log and continue
+        log.debug("Failed to mount {} ({}) to {} with flags {}: {}".
+                  format(source, fs, target, flags,
+                         os.strerror(ctypes.get_errno())))
+
+
+def umount(target):
+    if ctypes.CDLL(ctypes.util.find_library("c"), use_errno=True
+                   ).umount(target) < 0:
+        # log and continue
+        log.debug("Failed to umount {}: {}".
```

Comment:
log.error and for the rest of this file...

Reply:
Done.